app/components/TrackCard/tests/index.test.js

```diff
@@ -6,17 +6,27 @@
 
 import React from 'react';
 import TrackCard from '../index';
-import { renderWithIntl } from '@app/utils/testUtils';
+import { renderWithIntl, timeout } from '@app/utils/testUtils';
+import { fireEvent } from '@testing-library/dom';
 import { translate } from '@app/components/IntlGlobalProvider/index';
+import { BrowserRouter } from 'react-router-dom';
 
 describe('<TrackCard />', () => {
   it('should render and match with the snapshot', () => {
-    const baseComponent = renderWithIntl(<TrackCard />);
+    const baseComponent = renderWithIntl(
+      <BrowserRouter>
+        <TrackCard />
+      </BrowserRouter>
+    );
```

Comment:
Use [MemoryRouter](https://v5.reactrouter.com/web/api/MemoryRouter). Put that in `renderWithIntl`. You won't have to wrap your component with router everywhere.